### ISSUE TITLE
CoinCatch: fetch_balance() for perp. futures market returns zero for all coins

### ISSUE THREAD

**Issue by AminSaqi** [ccxt/ccxt#24239]
### Programming Languages

Python

### CCXT Version

4.4.28

### Description

I initialize the exchange like this:

`coincatch = ccxt.coincatch({
        "apiKey": "xxx",
        "secret": "yyy",
        "password": "zzz",
        "options": {
            "defaultType": "swap",
        },
    })`

Then I call `fetch_balance` function:

` balance = coincatch.fetch_balance()`

And every coin has ZERO amounts in all properties (free, used, total). Also, all amounts in response `info` is zero:

`{
  'coinId': '2', 
  'coinName': 'USDT', 
  'coinDisplayName': 'USDT',
   'available': '0',
   'frozen': '0', 
  'lock': '0', 
  'uTime': '1731432126123'
}`

However, I CAN open positions with the `create_order` method and it proofs that I have USDT deposited into the futures account.

### Code

```
  

```

**Comment by Dan-krm** [comment]
Hi @AminSaqi fetchBalance has an exchange specific `'productType'` parameter for swap fetchBalance that is defaulted to `'umcbl'`, could try setting the parameter to `'dmcbl'` and see if that works for you?

`balance = coincatch.fetch_balance(params={ "productType": "dmcbl"})`

You can also set it in the options alongside defaultType

**Comment by carlosmiei** [comment]
@AminSaqi Yeah we found a minor issue there and `defaultType` is not correctly parsed inside `fetchBalance`.

We will fix it shortly, or you can provide `type:'swap'` in `params` as a workaround.

**Comment by kroitor** [comment]
Hello @AminSaqi 👋

Closing this during a cleanup — no longer applicable, as the `coincatch` integration was removed from ccxt in #28346, so this balance issue can't be addressed within the library anymore. Thanks for the report! 🙏
